File: medium/11. Container With Most Water/11. Container With Most Water.ts
function maxArea(height: number[]): number {
    let left = 0, right = height.length - 1
    let maxSquare = 0

    while (left < right) {
        maxSquare = Math.max(maxSquare, Math.min(height[left], height[right]) * (right - left))

        if (height[left] < height[right])
            left += 1
        else
            right -= 1
    }

    return maxSquare
};

export { }

File: medium/11. Container With Most Water/11. Container With Most Water.go
package main

func maxArea(height []int) int {
	amount := 0

	first, second := 0, len(height)-1

	for first < second {
		boundaries := min(height[first], height[second])
		weight := len(height) - (first + second)
		amount = max(amount, boundaries*weight)

		if height[first] <= height[second] {
			first += 1
		} else {
			second -= 1
		}
	}

	return amount
}
